### ISSUE TITLE
Paypal doesn't work

### ISSUE THREAD

**Issue by tangcslanyu** [inspireui/support#3568]
Hi Sir
I have an PayPal configuration issue. Here is the screenshot in the attachment. 
I did exactly like your guide document. But meeting this error. The app and flutter are all newest version. 
<img width="355" alt="截屏2020-06-02 下午11 52 15" src="https://user-images.githubusercontent.com/30542660/83620682-3cc3f300-a5c0-11ea-8372-6e542922d495.png">

**Comment by tangcslanyu** [comment]
hello sir. Any news?

**Comment by minhcasi** [comment]
@tangcslanyu 

We wanted to thank you for your patience during this busy time. We currently have reduced staff and a significantly higher volume of inquiries. We’re working as hard as we can to get to your request as soon as possible.

We sincerely apologize for the delay and hope you’re staying safe.

@minhtruong315 - pls help for checking

**Comment by tangcslanyu** [comment]
I just type a US shipping address for testing, and choice free shipping. Then it turns out the Payment issue.

**Comment by minhtruong315** [comment]
Hi @tangcslanyu please try this solution
- Replace fromMagentoJson function on models/shipping_method.dart
```
ShippingMethod.fromMagentoJson(Map<String, dynamic> parsedJson) {
    id = parsedJson["carrier_code"];
    methodId = parsedJson["method_code"];
    title = parsedJson["carrier_title"];
    description = parsedJson["method_title"];
    cost = parsedJson["amount"] != null
        ? double.parse("${parsedJson["amount"]}")
        : 0.0;
  }
```
- Replace getPaymentMethods function on services/magento.dart
```
@override
  Future<List<PaymentMethod>> getPaymentMethods(
      {Address address, ShippingMethod shippingMethod, String token}) async {
    try {
      final params = {
        "addressInformation": {
          "shipping_address": address.toMagentoJson()["address"],
          "billing_address": address.toMagentoJson()["address"],
          "shipping_carrier_code": shippingMethod.id,
          "shipping_method_code": shippingMethod.methodId
        }
      };
      String url = token != null
          ? MagentoHelper.buildUrl(domain, "carts/mine/shipping-information")
          : MagentoHelper.buildUrl(
              domain, "guest-carts/$guestQuoteId/shipping-information");
      final res = await http.post(url,
          body: convert.jsonEncode(params),
          headers: token != null
              ? {
                  'Authorization': 'Bearer ' + token,
                  "content-type": "application/json"
                }
              : {"content-type": "application/json"});

      final body = convert.jsonDecode(res.body);
      if (res.statusCode == 200) {
        List<PaymentMethod> list = [];
        for (var item in body["payment_methods"]) {
          list.add(PaymentMethod.fromMagentoJson(item));
        }
        return list;
      } else if (body["message"] != null) {
        throw Exception(MagentoHelper.getErrorMessage(body));
      } else {
        throw Exception("Can not get payment methods");
      }
    } catch (err) {
      rethrow;
    }
  }
```
Then try again and let me know the result
Thanks

**Comment by tangcslanyu** [comment]
hello @minhtruong315 .
Thanks for your Answer.
After doing this I meet this error and can not load the app.

Compiler message:
lib/services/magento.dart:523:31: Error: 'getPaymentMethods' is already declared in this scope.
  Future<List<PaymentMethod>> getPaymentMethods(
                              ^^^^^^^^^^^^^^^^^
lib/services/magento.dart:484:31: Context: Previous declaration of 'getPaymentMethods'.
  Future<List<PaymentMethod>> getPaymentMethods(
                              ^^^^^^^^^^^^^^^^^

**Comment by minhtruong315** [comment]
Hi @tangcslanyu pls replace current function, dont add more the function.

**Comment by tangcslanyu** [comment]
Yes, I've replace the code, not just add it into. But get this error.

**Comment by minhtruong315** [comment]
@tangcslanyu can u upload your source code to github then invite me: minhtruong315@gmail.com
I will help you check on your source code.
Thanks

**Comment by tangcslanyu** [comment]
@minhtruong315  OK, which folder you want me to upload? Or you want all of them?

**Comment by minhtruong315** [comment]
Please upload all.

**Comment by tangcslanyu** [comment]
I don't know how to upload code here. I just send you via wetransfer to your email. Pls have a check.

**Comment by minhtruong315** [comment]
Hi @tangcslanyu I checked your source. it has 2 getPaymentMethods functions
You didn't replace. you added more the function
<img width="1260" alt="Screen Shot 2020-06-12 at 2 01 46 PM" src="https://user-images.githubusercontent.com/10402500/84474705-7d54f800-acb5-11ea-8cca-e853986b508f.png">

**Comment by minhtruong315** [comment]
@tangcslanyu please use this file
[magento.dart.zip](https://github.com/inspireui/support/files/4769260/magento.dart.zip)

**Comment by tangcslanyu** [comment]
@minhtruong315 
I'm sorry I still can not load the app. here is the log. pls have a check.

Xcode build done.                                           12.2s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    /Users/tangcansheng/Documents/Fluxstore/build/ios/Debug-iphonesimulator/flutter_webview_plugin/flutter_webview_plugin.framework: resource fork, Finder information, or similar detritus not allowed
    Command CodeSign failed with a nonzero exit code
    /Users/tangcansheng/Documents/Fluxstore/build/ios/Debug-iphonesimulator/webview_flutter/webview_flutter.framework: resource fork, Finder information, or similar detritus not allowed
    Command CodeSign failed with a nonzero exit code
    /Users/tangcansheng/Documents/Fluxstore/build/ios/Debug-iphonesimulator/wakelock/wakelock.framework: resource fork, Finder information, or similar detritus not allowed
    Command CodeSign failed with a nonzero exit code
    /Users/tangcansheng/Documents/Fluxstore/build/ios/Debug-iphonesimulator/url_launcher/url_launcher.framework: resource fork, Finder information, or similar detritus not allowed
    Command CodeSign failed with a nonzero exit code
    leveldb-library-eybtajssqogbglelzczlonrusore
    note: Using new build system
    note: Building targets in parallel
    note: Planning build
    note: Constructing build description

Could not build the application for the simulator.
Error launching application on iPhone SE (2nd generation).

**Comment by minhtruong315** [comment]
Hi @tangcslanyu add my skype: minhtruong315
I will support you via teamviewer.
Thanks

**Comment by minhtruong315** [comment]
Hi, I'm closing this ticket as there is no response after more than 2 days, if you have any questions or reproducing the issue kindly reopen or create a new ticket - thank you!
